tac/governing-documents/project-lifecycle.md

```diff
@@ -142,3 +142,128 @@ state independent of its prior history.
 [Proposal Template]: https://lf-decentralized-trust.github.io/project-proposals
+
+
+# Lifecycle Transition Guidelines and Recommendations
+
+The TAC must evaluate a project's lifecycle state during its [annual review](./project-annual-review.md) and make one of the following decisions:
+- Maintain its state
+- Upgrade (e.g., *Incubation* to *Graduated*)
+- Downgrade (e.g., *Graduated* to *Dormant*)
+
+In addition, a project's lifecycle state should be evaluated during the quarterly updates if the TAC sees that a project is struggling to meet the requirements of its current state.
+
+(*Note*: as a rough hierarchy, the *Graduated* state represents the highest maturity, followed by *Incubated*, followed by *Dormant*, followed by *Archived*.)
+
+The following criteria MUST be considered for each project when making one of the aforementioned decisions:
+- Legal
+- Diversity
+- Releases
+- Testing and Q/A
+- Security
+- Structure
+- Maintenance
+- Production
+- Documentation
+
+## Description and Breakup of Criteria
+
+Below is a guide for evaluating each of these criteria, with break-ups into sub-criteria that can be directly evaluated using one of three methods:
+1. Checking their OpenSSF Scorecard scores
+2. Checking their LFX Insights Best Practice Scores (produced by CLOMonitor)
+3. Manual inspection of code repositories
+
+(*Note*: The term "close to" below implies that a high (or highest) score is not mandatory but highly desired. If a project falls short on some such criteria but is otherwise mature and well-functioning, the TAC may ignore those criteria while encouraging the project's maintainers to strive for a higher score.)
+
+### Legal Criteria
+
+- (OpenSSF Score): *License*: MUST be 10
+
+### Diversity Criteria
+
+- (OpenSSF Score): *Contributors*: SHOULD be close to 10
+- (Manual): the `MAINTAINERS.md` files in the project's repositories MUST collectively indicate the involvement of more than one organization
+
+- Evaluation:
+  * To be a *Graduated* project, the number of organizations from which maintainers are drawn must be at least 3, and the OpenSSF Score SHOULD be 10.
+  * To be an *Incubated* project, the number of organizations from which maintainers are drawn must be at least 2.
+
+### Releases
+
+- (OpenSSF Score): *Packaging*: SHOULD be 10
+- (Manual): the number and frequency of releases in the recent past (quarter/year), as identified by inspecting the project's repositories, must be reasonably high (at least for one of the repositories)
+
+### Testing and Q/A
+
+- (OpenSSF Score): *CI Tests*: SHOULD be 10
+
+### Security
+
+- (OpenSSF Score): *Dangerous Workflow*: MUST be 10
+- (OpenSSF Score): *Token Permissions*: MUST be 10
+- (OpenSSF Score): *Branch-Protection*: MUST be 9 (Tier 4)
+- (OpenSSF Score): *Dependency-Update-Tools*: MUST be 10
+- (OpenSSF Score): *Fuzzing*: SHOULD be close to 10
+- (OpenSSF Score): *Pinned-Dependencies*: SHOULD NOT be 0
```

Comment:
I agree in theory, but in practice it may be zero.
For example, for Fabric the most important thing is that Go dependencies are pinned. Fabric does indeed pin Go dependencies. However the Scorecard doesn't seem to pick that up. Fabric has decided not to pin other dependencies such as Github Actions, since we like to use the latest and greatest actions when running CI to see if anything breaks without having to invest in constant maintenance. The Scorecard dings us for the Github Actions and gives Fabric a 0.

Reply:
Agreed, deviations should be highlighted and discussed during the annual reviews.